<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html xmlns:cis="http://zlab.bu.edu/schema/cisml" xmlns:fimo="http://noble.gs.washington.edu/schema/cisml" xmlns:mem="http://noble.gs.washington.edu/meme">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="UTF-8">
<title>FIMO Results</title>
<style type="text/css">
td.left {text-align: left;}
td.right {text-align: right; padding-right: 1cm;}
</style>
</head>
<body color="#000000">
<a name="top_buttons"></a>
<hr>
<table summary="buttons" align="left" cellspacing="0">
<tr>
<td bgcolor="#00FFFF"><a href="#database_and_motifs"><b>Database and Motifs</b></a></td>
<td bgcolor="#DDFFDD"><a href="#sec_i"><b>High-scoring Motif Occurences</b></a></td>
<td bgcolor="#DDDDFF"><a href="#debugging_information"><b>Debugging Information</b></a></td>
<td bgcolor="#FFDD00"><a href="fimo.tsv""><b>Results in TSV Format</b></a></td>
<td bgcolor="#FFDDFF"><a href="fimo.gff""><b>Results in GFF3 Format</b></a></td>
</tr>
</table>
<br/>
<br/>
<hr/>
<center><big><b>FIMO - Motif search tool</b></big></center>
<hr>
<p>
FIMO version 5.3.0, (Release date: Wed Nov 25 19:19:29 2020 -0800)
</p>
<p>
For further information on how to interpret these results please access <a href="http://meme-suite.org/doc/fimo-output-format.html">http://meme-suite.org/doc/fimo-output-format.html</a>.<br>
To get a copy of the FIMO software please access <a href="http://meme-suite.org">http://meme-suite.org</a>

</p>
<p>If you use FIMO in your research, please cite the following paper:<br>
Charles E. Grant, Timothy L. Bailey, and William Stafford Noble,
"FIMO: Scanning for occurrences of a given motif",
<i>Bioinformatics</i>, <b>27</b>(7):1017-1018, 2011.
<a href="http://bioinformatics.oxfordjournals.org/content/27/7/1017">[full text]</a></p>
<hr>
<center><big><b><a name="database_and_motifs">DATABASE AND MOTIFS</a></b></big></center>
<hr>
<div style="padding-left: 0.75in; line-height: 1em; font-family: monospace;">
<p>
  DATABASE /opt/project/data/reference/ORF7a/P0DTC7.fasta
  <br />
  Database contains 1 sequences, 121 residues
</p>
<p>
  MOTIFS /opt/project/data/motif/ORF7a/meme/meme.txt (Protein)
  <table>
    <thead>
      <tr>
        <th style="border-bottom: 1px dashed;">MOTIF</th>
        <th style="border-bottom: 1px dashed; padding-left: 1em;">WIDTH</th>
        <th style="border-bottom: 1px dashed; padding-left: 1em;text-align:left;" >
         BEST POSSIBLE MATCH
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td style="text-align:right;">YHYQECVRGTTVLLKEPCPSGTYEGNSPFHPLADNKFALTCTSTHFAFAC</td>
        <td style="text-align:right;padding-left: 1em;">50</td>
        <td style="text-align:left;padding-left: 1em;">YHYQECVRGTTVLLKEPCPSGTYEGNSPFHPLADNKFALTCTSTHFAFAC</td>
       </tr>
      <tr>
        <td style="text-align:right;">ADGTRHTYQLRARSVSPKLFIRQEEVHQE</td>
        <td style="text-align:right;padding-left: 1em;">29</td>
        <td style="text-align:left;padding-left: 1em;">ADGTRHTYQLRARSVSPKLFIRQEEVQQE</td>
       </tr>
      <tr>
        <td style="text-align:right;">FLIVAALVFIILCFTIKRKTE</td>
        <td style="text-align:right;padding-left: 1em;">21</td>
        <td style="text-align:left;padding-left: 1em;">FLIVAALVFIILCFTIKRKTE</td>
       </tr>
      <tr>
        <td style="text-align:right;">MKIILFLTLIALASC</td>
        <td style="text-align:right;padding-left: 1em;">15</td>
        <td style="text-align:left;padding-left: 1em;">MKIILFLTLIALASC</td>
       </tr>
    </tbody>
  </table>
</p>
<p>
Random model letter frequencies (--motif--):
<br/>

A 0.076 C 0.048 D 0.018 E 0.065 F 0.076 G 0.033 H 0.035 I 0.062 K 0.051 
L 0.127 M 0.010 N 0.018 P 0.049 Q 0.037 R 0.048 S 0.059 T 0.091 V 0.054 
W 0.002 Y 0.043 </p>
</div>
<hr>
<center><big><b><a name="sec_i">SECTION I: HIGH-SCORING MOTIF OCCURENCES</a></b></big></center>
<hr>
<ul>
<li>
There were 4 motif occurences with a p-value less than 0.0001.

The full set of motif occurences can be seen in the
TSV (tab-delimited values) output file
<a href="fimo.tsv">fimo.tsv</a>, 
the GFF3 file 
<a href="fimo.gff">fimo.gff</a> 
which may be suitable for uploading to the 
<a href="http://genome.ucsc.edu/cgi-bin/hgTables">UCSC Genome Table Browser</a>
(assuming the FASTA input sequences included genomic coordinates in UCSC or Galaxy format),
or the XML file 
<a href="fimo.xml">fimo.xml</a>.
</li>
<li>
The <i>p</i>-value of a motif occurrence is defined as the
probability of a random sequence of the same length as the motif
matching that position of the sequence with as good or better a score.
</li>
<li>
The score for the match of a position in a sequence to a motif
is computed by summing the appropriate entries from each column of
the position-dependent scoring matrix that represents the motif.
</li>
<li>
The q-value of a motif occurrence is defined as the
false discovery rate if the occurrence is accepted as significant.
</li>
<li>The table is sorted by increasing <i>p</i>-value.</li>
</ul>
<table border="1">
<thead>
<tr>
<th>Motif ID</th>
<th>Alt ID</th>
<th>Sequence Name</th>
<th>Strand</th>
<th>Start</th>
<th>End</th>
<th>p-value</th>
<th>q-value</th>
<th>Matched Sequence</th>
</tr>
</thead>
<tbody>
    <tr>
      <td style="text-align:left;">YHYQECVRGTTVLLKEPCPSGTYEGNSPFHPLADNKFALTCTSTHFAFAC</td>
      <td style="text-align:left;">MEME-1</td>
      <td style="text-align:left;">sp|P0DTC7|NS7A_SARS2</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">18</td>
      <td style="text-align:left;">67</td>
      <td style="text-align:left;">3.64e-62</td>
      <td style="text-align:left;">2.62e-60</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">YHYQECVRGTTVLLKEPCSSGTYEGNSPFHPLADNKFALTCFSTQFAFAC</td>
   </tr>
    <tr>
      <td style="text-align:left;">ADGTRHTYQLRARSVSPKLFIRQEEVHQE</td>
      <td style="text-align:left;">MEME-2</td>
      <td style="text-align:left;">sp|P0DTC7|NS7A_SARS2</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">68</td>
      <td style="text-align:left;">96</td>
      <td style="text-align:left;">9.33e-34</td>
      <td style="text-align:left;">8.68e-32</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">PDGVKHVYQLRARSVSPKLFIRQEEVQEL</td>
   </tr>
    <tr>
      <td style="text-align:left;">FLIVAALVFIILCFTIKRKTE</td>
      <td style="text-align:left;">MEME-3</td>
      <td style="text-align:left;">sp|P0DTC7|NS7A_SARS2</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">101</td>
      <td style="text-align:left;">121</td>
      <td style="text-align:left;">1.35e-23</td>
      <td style="text-align:left;">1.34e-21</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">FLIVAAIVFITLCFTLKRKTE</td>
   </tr>
    <tr>
      <td style="text-align:left;">MKIILFLTLIALASC</td>
      <td style="text-align:left;">MEME-4</td>
      <td style="text-align:left;">sp|P0DTC7|NS7A_SARS2</td>
      <td style="text-align:center;">+</td>
      <td style="text-align:left;">1</td>
      <td style="text-align:left;">15</td>
      <td style="text-align:left;">1.4e-16</td>
      <td style="text-align:left;">1.47e-14</td>
      <td style="text-align:left;font-size:x-large;font-family:monospace;">MKIILFLALITLATC</td>
   </tr>
</tbody>
</table>

<hr>
<center><big><b><a name="debugging_information">DEBUGGING INFORMATION</a></b></big></center>
<hr>
<p>
Command line:
</p>
<pre>
fimo --bfile --motif-- -oc /opt/project/data/motif/ORF7a/fimo /opt/project/data/motif/ORF7a/meme/meme.txt /opt/project/data/reference/ORF7a/P0DTC7.fasta
</pre>
<p>
Settings:
</p>
<pre>
<table>
  <tr>
    <td style="padding-right: 2em">output_directory = /opt/project/data/motif/ORF7a/fimo</td>
    <td style="padding-left: 5em; padding-right: 2em">MEME file name = /opt/project/data/motif/ORF7a/meme/meme.txt</td>
    <td style="padding-left: 5em; padding-right: 2em">sequence file name = /opt/project/data/reference/ORF7a/P0DTC7.fasta</td>
  </tr>  <tr>
    <td style="padding-right: 2em">background file name = --motif--</td>
    <td style="padding-left: 5em; padding-right: 2em">alphabet = Protein</td>
    <td style="padding-left: 5em; padding-right: 2em">max stored scores = 100000</td>
  </tr>  <tr>
    <td style="padding-right: 2em">allow clobber = true</td>
    <td style="padding-left: 5em; padding-right: 2em">compute q-values = true</td>
    <td style="padding-left: 5em; padding-right: 2em">parse genomic coord. = false</td>
  </tr>
  <tr>
    <td style="padding-right: 2em">text only = false</td>
    <td style="padding-left: 5em; padding-right: 2em">scan both strands = false</td>
    <td style="padding-left: 5em; padding-right: 2em">max strand = false</td>
  </tr>
  <tr>
    <td style="padding-right: 2em">threshold type = p-value</td>
    <td style="padding-left: 5em; padding-right: 2em">output theshold = 0.0001</td>
    <td style="padding-left: 5em; padding-right: 2em">pseudocount = 0.1</td>
  </tr>
  <tr>
    <td style="padding-right: 2em">alpha = 1</td>
    <td style="padding-left: 5em; padding-right: 2em">verbosity = 2</td>
    <td style="padding-left: 5em; padding-right: 2em"></td>
  </tr>

</table>
</pre>
<p>
This information can be useful in the event you wish to report a
problem with the FIMO software.
</p>
<hr>
<span style="background-color: #DDDDFF"><a href="#top_buttons"><b>Go to top</b></a></span>
</body>
</html>
