{
 "cells": [
  {
   "cell_type": "markdown",
   "source": [
    "# Identificação de Homólogos"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [],
   "source": [
    "import os\n",
    "from Bio import SeqIO\n",
    "from Bio.Blast import NCBIWWW, NCBIXML\n",
    "from scripts_python import BLASTer\n",
    "\n",
    "a = %pwd\n",
    "wd = (a.rsplit('/',2))[0]\n",
    "seq_id = \"NM_001114380\"\n",
    "prot_seq_id = \"P20701\"\n",
    "gene = \"ITGAL\"\n",
    "file_blast_ncbi = \"00D8F7PY016-Alignment.xml\"\n",
    "file_blast_uniprot = \"B20210131A94466D2655679D1FD8953E075198DA8030E49B.fasta\""
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Foram carregadas 100 seq.\n"
     ]
    }
   ],
   "source": [
    "result_handle= open(os.path.join( wd,\"data/homologue\", gene, file_blast_ncbi))\n",
    "blast_ncbi = NCBIXML.read(result_handle)\n",
    "result_handle.close()\n",
    "\n",
    "print('Foram carregadas {0} seq.'.format(len(blast_ncbi.alignments)))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Resultados do BLAST no NCBI"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accession Identity%\n",
      "NP_002200 1.0\n",
      "BAG36913 0.9991452991452991\n",
      "XP_005255370 0.9991452991452991\n",
      "CAA68747 0.9982905982905983\n",
      "NP_001029341 0.9923076923076923\n",
      "XP_034795955 0.9923076923076923\n",
      "PNI12205 0.9923076923076923\n",
      "XP_034795956 0.9914529914529915\n",
      "XP_024209187 0.9914529914529915\n",
      "EAW52244 0.9566639411283728\n",
      "AAC31672 0.955846279640229\n",
      "XP_024089734 0.9777777777777777\n",
      "XP_032011574 0.9709401709401709\n",
      "XP_024209188 0.976068376068376\n",
      "XP_030653892 0.9675213675213675\n",
      "XP_024209189 0.9675213675213675\n",
      "XP_023046865 0.958974358974359\n",
      "XP_023046867 0.958904109589041\n",
      "XP_023046866 0.9581196581196582\n",
      "XP_010386254 0.9555555555555556\n",
      "EHH31590 0.9538461538461539\n",
      "XP_025225979 0.9555555555555556\n",
      "XP_037845568 0.9573742540494459\n",
      "XP_009197106 0.9564102564102565\n",
      "XP_033074013 0.9512820512820512\n",
      "XP_014981430 0.9529914529914529\n",
      "EHH60316 0.9511986301369864\n",
      "XP_033074015 0.9511986301369864\n",
      "XP_033074014 0.9504273504273504\n",
      "XP_017736913 0.9512820512820512\n",
      "XP_028697390 0.9253112033195021\n",
      "XP_011544151 1.0\n",
      "XP_034795957 0.9916201117318436\n",
      "XP_024209192 0.9916201117318436\n",
      "XP_021788969 0.9307692307692308\n",
      "XP_011837937 0.9122807017543859\n",
      "XP_032099091 0.9094017094017094\n",
      "XP_035123298 0.9025641025641026\n",
      "XP_012319913 0.9\n",
      "XP_003930122 0.9008547008547009\n",
      "XP_024209194 0.991270611057226\n",
      "XP_023046868 0.957169459962756\n",
      "XP_024648899 0.8936535162950258\n",
      "XP_009197108 0.9543761638733705\n",
      "XP_006721107 0.9949186991869918\n",
      "XP_006721107 1.0\n",
      "NP_001107852 0.9939024390243902\n",
      "NP_001107852 1.0\n",
      "AAP88867 0.9939024390243902\n",
      "AAP88867 1.0\n",
      "XP_024209190 0.9857723577235772\n",
      "XP_024209190 1.0\n",
      "XP_024209191 0.9847560975609756\n",
      "XP_024209191 1.0\n",
      "XP_021788970 0.9553831231813773\n",
      "XP_024089735 0.9705284552845529\n",
      "XP_024089735 0.9724770642201835\n",
      "XP_032099093 0.909683426443203\n",
      "XP_008592140 0.8222222222222222\n",
      "XP_014697888 0.82066609735269\n",
      "XP_014585502 0.8189581554227157\n",
      "XP_008512525 0.8198121263877028\n",
      "XP_030653900 0.9603658536585366\n",
      "XP_030653900 0.963302752293578\n",
      "XP_008055949 0.8469565217391304\n",
      "XP_012515779 0.8304794520547946\n",
      "XP_014697889 0.8198121263877028\n",
      "XP_005598832 0.8181041844577285\n",
      "XP_020136933 0.8277634961439588\n",
      "XP_012619680 0.8269065981148244\n",
      "XP_027468348 0.8065068493150684\n",
      "XP_025709990 0.8056506849315068\n",
      "XP_004397279 0.8047945205479452\n",
      "XP_012619677 0.8138163437236732\n",
      "XP_025709743 0.8047945205479452\n",
      "XP_027966076 0.8039383561643836\n",
      "XP_033074018 0.9502982107355865\n",
      "XP_036683247 0.8193493150684932\n",
      "XP_027468350 0.8056506849315068\n",
      "XP_025709991 0.8047945205479452\n",
      "XP_012619678 0.8129738837405224\n",
      "XP_025709744 0.8039383561643836\n",
      "XP_027966077 0.803082191780822\n",
      "XP_036683248 0.8184931506849316\n",
      "XP_006914189 0.8096054888507719\n",
      "TEA38802 0.8176369863013698\n",
      "XP_026982189 0.8159246575342466\n",
      "XP_033695841 0.8176369863013698\n",
      "XP_011376968 0.8087478559176673\n",
      "XP_030727609 0.8167808219178082\n",
      "XP_032286925 0.8107876712328768\n",
      "XP_024600432 0.8159246575342466\n",
      "XP_032460125 0.8159246575342466\n",
      "XP_012388162 0.815068493150685\n",
      "XP_006914188 0.8089117395029991\n",
      "XP_007186469 0.8159246575342466\n",
      "XP_026982190 0.815068493150685\n",
      "XP_033695842 0.8167808219178082\n",
      "XP_011376967 0.8080548414738646\n",
      "XP_034844757 0.8082191780821918\n",
      "XP_029075724 0.8125\n",
      "XP_030727610 0.8159246575342466\n",
      "XP_032286926 0.809931506849315\n",
      "XP_012388163 0.8142123287671232\n",
      "XP_034844758 0.8073630136986302\n",
      "Total: 105\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import SeqIO\n",
    "import re\n",
    "\n",
    "query_len = blast_ncbi.query_length\n",
    "homologue = []\n",
    "sources = {}\n",
    "\n",
    "print(\"Accession\", \"Identity%\")\n",
    "for aln in blast_ncbi.alignments:\n",
    "    for HSP in aln.hsps:\n",
    "        if HSP.score > 80 and HSP.expect < 1.0e-10 and HSP.identities/HSP.align_length > 0.80:\n",
    "            print(aln.accession,HSP.identities/HSP.align_length)\n",
    "            species = re.findall('\\[[^\\]]*\\]', aln.hit_def)[0].replace(\"[\",\"\").replace(\"]\",\"\")\n",
    "            if species not in sources:\n",
    "                sources[species] = 1\n",
    "            else:\n",
    "                sources[species] +=1\n",
    "            homologue.append(SeqRecord(Seq(HSP.sbjct), aln.title.split( \">\", 1)[0], \"\", \"\"))\n",
    "\n",
    "\n",
    "print(\"Total:\",len(homologue))"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "source": [
    "for source in sources.keys():\n",
    "    print(source, \":\", sources[source])\n",
    "\n",
    "SeqIO.write(homologue, os.path.join(wd,\"data/homologue\", gene, f\"{seq_id}_{gene}.fasta\"), \"fasta\")"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "execution_count": 12,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Homo sapiens : 11\n",
      "Pan troglodytes : 11\n",
      "Pan paniscus : 3\n",
      "Pongo abelii : 3\n",
      "Hylobates moloch : 1\n",
      "Nomascus leucogenys : 3\n",
      "Piliocolobus tephrosceles : 4\n",
      "Rhinopithecus roxellana : 1\n",
      "Macaca mulatta : 3\n",
      "Theropithecus gelada : 1\n",
      "Chlorocebus sabaeus : 1\n",
      "Papio anubis : 4\n",
      "Trachypithecus francoisi : 4\n",
      "Macaca fascicularis : 1\n",
      "Rhinopithecus bieti : 1\n",
      "Mandrillus leucophaeus : 1\n",
      "Sapajus apella : 2\n",
      "Callithrix jacchus : 1\n",
      "Aotus nancymaae : 1\n",
      "Saimiri boliviensis boliviensis : 1\n",
      "Macaca nemestrina : 1\n",
      "synthetic construct : 2\n",
      "Galeopterus variegatus : 1\n",
      "Equus asinus : 2\n",
      "Equus caballus : 2\n",
      "Equus przewalskii : 1\n",
      "Carlito syrichta : 1\n",
      "Propithecus coquereli : 1\n",
      "Microcebus murinus : 4\n",
      "Zalophus californianus : 2\n",
      "Callorhinus ursinus : 4\n",
      "Odobenus rosmarus divergens : 1\n",
      "Eumetopias jubatus : 2\n",
      "Balaenoptera musculus : 2\n",
      "Pteropus alecto : 2\n",
      "Sousa chinensis : 1\n",
      "Lagenorhynchus obliquidens : 2\n",
      "Tursiops truncatus : 2\n",
      "Pteropus vampyrus : 2\n",
      "Globicephala melas : 2\n",
      "Phoca vitulina : 2\n",
      "Neophocaena asiaeorientalis asiaeorientalis : 1\n",
      "Phocoena sinus : 1\n",
      "Orcinus orca : 2\n",
      "Balaenoptera acutorostrata scammoni : 1\n",
      "Mirounga leonina : 2\n",
      "Monodon monoceros : 1\n"
     ]
    },
    {
     "data": {
      "text/plain": "105"
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ]
  },
  {
   "cell_type": "markdown",
   "source": [
    "### Resultados UNIPROT\n",
    "\n",
    "#### Comentários"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%% md\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "outputs": [],
   "source": [
    "path = os.path.join(wd, \"data/homologue\", gene, file_blast_uniprot)\n",
    "blast_uniprot_raw = SeqIO.parse(path, format=\"fasta\")\n",
    "blast_uniprot = []\n",
    "blast_uniprot_id = []\n",
    "\n",
    "for protein in blast_uniprot_raw:\n",
    "    blast_uniprot.append(protein)\n",
    "    seq_id = re.findall('\\|[^|]*\\|', protein.id)[0].replace(\"|\",\"\")\n",
    "    blast_uniprot_id.append(seq_id)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "outputs": [],
   "source": [
    "from wget import download\n",
    "\n",
    "for accession in blast_uniprot_id:\n",
    "    url = \"https://www.uniprot.org/uniprot/{0}.xml\".format(accession)\n",
    "    path = os.path.join(wd, \"data/homologue\", gene, \"uniprot\",\"{0}.xml\".format(accession))\n",
    "    download(url, path)\n"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "function\n",
      "Integrin ITGAL/ITGB2 is a receptor for ICAM1, ICAM2, ICAM3 and ICAM4. Integrin ITGAL/ITGB2 is a receptor for F11R (PubMed:11812992, PubMed:15528364). Integin ITGAL/ITGB2 is a receptor for the secreted form of ubiquitin-like protein ISG15; the interaction is mediated by ITGAL (PubMed:29100055). Involved in a variety of immune phenomena including leukocyte-endothelial cell interaction, cytotoxic T-cell mediated killing, and antibody dependent killing by granulocytes and monocytes. Contributes to natural killer cell cytotoxicity (PubMed:15356110). Involved in leukocyte adhesion and transmigration of leukocytes including T-cells and neutrophils (PubMed:11812992). Required for generation of common lymphoid progenitor cells in bone marrow, indicating a role in lymphopoiesis (By similarity). Integrin ITGAL/ITGB2 in association with ICAM3, contributes to apoptotic neutrophil phagocytosis by macrophages (PubMed:23775590). : P20701\n",
      "Integrin ITGAL/ITGB2 is a receptor for ICAM1, ICAM2, ICAM3 and ICAM4. Integrin ITGAL/ITGB2 is a receptor for F11R. Integrin ITGAL/ITGB2 is a receptor for the secreted form of ubiquitin-like protein ISG15; the interaction is mediated by ITGAL. Involved in a variety of immune phenomena including leukocyte-endothelial cell interaction, cytotoxic T-cell mediated killing, and antibody dependent killing by granulocytes and monocytes. Contributes to natural killer cell cytotoxicity. Involved in leukocyte adhesion and transmigration of leukocytes including T-cells and neutrophils. Required for generation of common lymphoid progenitor cells in bone marrow, indicating the role in lymphopoiesis. Integrin ITGAL/ITGB2 in association with ICAM3, contributes to apoptotic neutrophil phagocytosis by macrophages. : P61625\n",
      "Integrin ITGAL/ITGB2 is a receptor for ICAM1, ICAM2, ICAM3 and ICAM4 (PubMed:2051027). Integrin ITGAL/ITGB2 is a receptor for F11R (By similarity). Integrin ITGAL/ITGB2 is a receptor for the secreted form of ubiquitin-like protein ISG15; the interaction is mediated by ITGAL (PubMed:29100055). Involved in a variety of immune phenomena including leukocyte-endothelial cell interaction, cytotoxic T-cell mediated killing, and antibody dependent killing by granulocytes and monocytes. Contributes to natural killer cell cytotoxicity. Involved in leukocyte adhesion and transmigration of leukocytes including T-cells and neutrophils (PubMed:16234355, PubMed:24158516). Required for generation of common lymphoid progenitor cells in bone marrow, indicating the role in lymphopoiesis (PubMed:25108025). Integrin ITGAL/ITGB2 in association with ICAM3, contributes to apoptotic neutrophil phagocytosis by macrophages. : P24063\n",
      "Integrin ITGAM/ITGB2 is implicated in various adhesive interactions of monocytes, macrophages and granulocytes as well as in mediating the uptake of complement-coated particles and pathogens (PubMed:9558116, PubMed:20008295). It is identical with CR-3, the receptor for the iC3b fragment of the third complement component. It probably recognizes the R-G-D peptide in C3b. Integrin ITGAM/ITGB2 is also a receptor for fibrinogen, factor X and ICAM1. It recognizes P1 and P2 peptides of fibrinogen gamma chain. Regulates neutrophil migration (PubMed:28807980). In association with beta subunit ITGB2/CD18, required for CD177-PRTN3-mediated activation of TNF primed neutrophils (PubMed:21193407). May regulate phagocytosis-induced apoptosis in extravasated neutrophils (By similarity). May play a role in mast cell development (By similarity). Required with TYROBP/DAP12 in microglia to control production of microglial superoxide ions which promote the neuronal apoptosis that occurs during brain development (By similarity). : P11215\n",
      "Integrin ITGAM/ITGB2 is implicated in various adhesive interactions of monocytes, macrophages and granulocytes as well as in mediating the uptake of complement-coated particles and pathogens (By similarity). It is identical with CR-3, the receptor for the iC3b fragment of the third complement component. It probably recognizes the R-G-D peptide in C3b. Integrin ITGAM/ITGB2 is also a receptor for fibrinogen, factor X and ICAM1. It recognizes P1 and P2 peptides of fibrinogen gamma chain. Regulates neutrophil migration. In association with beta subunit ITGB2/CD18, required for CD177-PRTN3-mediated activation of TNF primed neutrophils (By similarity). May regulate phagocytosis-induced apoptosis in extravasated neutrophils (By similarity). May play a role in mast cell development (By similarity). Required with TYROBP/DAP12 in microglia to control production of microglial superoxide ions which promote the neuronal apoptosis that occurs during brain development (PubMed:18685038). : P05555\n",
      "Integrin alpha-X/beta-2 is a receptor for fibrinogen. It recognizes the sequence G-P-R in fibrinogen. It mediates cell-cell interaction during inflammatory responses. It is especially important in monocyte adhesion and chemotaxis (By similarity). : Q9QXH4\n",
      "Integrin alpha-X/beta-2 is a receptor for fibrinogen. It recognizes the sequence G-P-R in fibrinogen. It mediates cell-cell interaction during inflammatory responses. It is especially important in monocyte adhesion and chemotaxis. : P20702\n",
      "Integrin alpha-D/beta-2 is a receptor for ICAM3 and VCAM1. May play a role in the atherosclerotic process such as clearing lipoproteins from plaques and in phagocytosis of blood-borne pathogens, particulate matter, and senescent erythrocytes from the blood (By similarity). : Q9QYE7, Q3V0T4\n",
      "Integrin alpha-D/beta-2 is a receptor for ICAM3 and VCAM1. May play a role in the atherosclerotic process such as clearing lipoproteins from plaques and in phagocytosis of blood-borne pathogens, particulate matter, and senescent erythrocytes from the blood. : Q13349\n",
      "Integrin alpha-11/beta-1 is a receptor for collagen. : Q9UKX5, P61622\n",
      "Integrin alpha-10/beta-1 is a receptor for collagen. : O75578\n",
      "\n",
      "\n",
      "subunit\n",
      "Heterodimer of an alpha and a beta subunit (PubMed:12526797). The ITGAL alpha subunit associates with the ITGB2 beta subunit (PubMed:12526797). Interacts with THBD (PubMed:27055590). : P20701\n",
      "Heterodimer of an alpha and a beta subunit. The ITGAL alpha subunit associates with the ITGB2 beta subunit. Interacts with THBD. : P61625, P24063\n",
      "Heterodimer of an alpha and a beta subunit. ITGAM associates with ITGB2. Found in a complex with CD177 and ITGB2/CD18 (PubMed:21193407). Interacts with JAM3 (PubMed:15194813, PubMed:12208882). Interacts with THBD (PubMed:27055590). Interacts with complement factor H/CFH; this interaction mediates adhesion of neutrophils to pathogens leading to pathogen clearance. : P11215\n",
      "Heterodimer of an alpha and a beta subunit. ITGAM associates with ITGB2. Found in a complex with CD177 and ITGB2/CD18. Interacts with JAM3. Interacts with THBD. Interacts with complement factor H/CFH; this interaction mediates adhesion of neutrophils to pathogens leading to pathogen clearance. : P05555\n",
      "Heterodimer of an alpha and a beta subunit. Alpha-X associates with beta-2 (By similarity). : Q9QXH4\n",
      "Heterodimer of an alpha and a beta subunit. Alpha-X associates with beta-2. : P20702\n",
      "Heterodimer of an alpha and a beta subunit. Alpha-D associates with beta-2 (By similarity). : Q9QYE7, Q3V0T4\n",
      "Heterodimer of an alpha and a beta subunit. Alpha-D associates with beta-2. : Q13349\n",
      "Heterodimer of an alpha and a beta subunit. Alpha-11 associates with beta-1. Interacts with RAB21. : Q9UKX5\n",
      "Heterodimer of an alpha and a beta subunit. Alpha-10 associates with beta-1. : O75578\n",
      "Heterodimer of an alpha and a beta subunit. Alpha-11 associates with beta-1 (By similarity). Interacts with RAB21 (By similarity). : P61622\n",
      "\n",
      "\n",
      "interaction_intactId\n",
      "EBI-961214 : P20701\n",
      "EBI-11616611 : P61625\n",
      "EBI-2568251 : P11215\n",
      "EBI-2568308 : P20702\n",
      "\n",
      "\n",
      "subcellularlocation_location\n",
      "Cell membrane : P20701, P61625, P24063, P11215, P05555\n",
      "Membrane : Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "\n",
      "\n",
      "subcellularlocation_topology\n",
      "Single-pass type I membrane protein : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "\n",
      "\n",
      "alternativeproducts_isoform\n",
      "P20701-1 : P20701\n",
      "P11215-1 : P11215\n",
      "P05555-1 : P05555\n",
      "Q9UKX5-1 : Q9UKX5\n",
      "O75578-1 : O75578\n",
      "\n",
      "\n",
      "tissuespecificity\n",
      "Leukocytes. : P20701, P24063\n",
      "Predominantly expressed in monocytes and granulocytes (PubMed:1346576). Expressed in neutrophils (at protein level) (PubMed:21193407). : P11215\n",
      "Predominantly expressed in monocytes and granulocytes (PubMed:3887182, PubMed:8986723). Expressed in a subset of peritoneal mast cells (PubMed:9862668). Expressed in microglia (at protein level) (PubMed:18685038). : P05555\n",
      "Predominantly expressed in monocytes and granulocytes. : P20702\n",
      "Expressed moderately on myelomonocytic cell lines and subsets of peripheral blood leukocytes and strongly on tissue-specialized cells, including macrophages foam cells within atherosclerotic plaques, and on splenic red pulp macrophages. : Q13349\n",
      "According to PubMed:10464311, highest levels of expression in uterus and heart, intermediate levels in skeletal muscle and intermediate to low levels in pancreas, kidney and placenta. According to PubMed:10486209, also found in brain, colon, lung, small intestine, stomach, testis, salivary glands, thyroid glands and prostate. Very low levels in peripheral blood lymphocytes, fetal brain and fetal liver. : Q9UKX5\n",
      "Widely expressed with highest expression in muscle and heart. Found in articular cartilage. : O75578\n",
      "\n",
      "\n",
      "domain\n",
      "The integrin I-domain (insert) is a VWFA domain (PubMed:2537322). Integrins with I-domains do not undergo protease cleavage. The I-domain is necessary and sufficient for interaction with ICAM1 and F11R (PubMed:15528364). : P20701\n",
      "The integrin I-domain (insert) is a VWFA domain. Integrins with I-domains do not undergo protease cleavage. The I-domain is necessary and sufficient for interaction with ICAM1 and F11R. : P61625, P24063\n",
      "The integrin I-domain (insert) is a VWFA domain. Integrins with I-domains do not undergo protease cleavage. : P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "\n",
      "\n",
      "PTM\n",
      "In resting T-cells, up to 40% of surface ITGAL is constitutively phosphorylated. Phosphorylation causes conformational changes needed for ligand binding and is necessary for activation by some physiological agents. : P20701\n",
      "In resting T-cells, up to 40% of surface ITGAL is constitutively phosphorylated. Phosphorylation causes conformational changes needed for ligand binding and is necessary for the activation by some physiological agents. : P61625, P24063\n",
      "\n",
      "\n",
      "similarity\n",
      "Belongs to the integrin alpha chain family. : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "\n",
      "\n",
      "onlineinformation\n",
      "NIEHS-SNPs@http://egp.gs.washington.edu/data/itgal/ : P20701\n",
      "Atlas of Genetics and Cytogenetics in Oncology and Haematology@http://atlasgeneticsoncology.org/Genes/ITGA11ID41001ch15q23.html : Q9UKX5\n",
      "\n",
      "\n",
      "disruptionphenotype\n",
      "Mice show decreased cellularity in thymus but not spleen, and impaired early T cell development (PubMed:25108025). Obese mutant mice show decreased total number of T-cells, lower levels of neutrophil elastase and reduced cytotoxic T-cell proliferation in adipose tissue, as well as improved glucose tolerance and insulin resistance in comparison to obese wild type mice (PubMed:24158516). Splenocytes from mutant mice do not respond to extracellular Isg15 as demonstrated by lack of Ifng secretion in contrast to wild-type cells which secrete Ifng in response to treatment with Isg15 (PubMed:29100055). : P24063\n",
      "Mice are born at the expected Mendelian rate (PubMed:8986723). The numbers of blood leukocytes are normal and neutrophils rolling on blood vessels is not affected (PubMed:8986723). Impaired LTB4-induced neutrophil adhesion to venules (PubMed:8986723). Numbers of mast cells in the peritoneal cavity, peritoneal wall and in the dorsal skin are reduced (PubMed:9862668). In an experimental model of thioglycolate-induced peritonitis, abnormal accumulation of recruited neutrophils due to impaired apoptosis (PubMed:8986723). In an experimental model of acute septic peritonitis, 5-fold increase in mortality rate, reduced neutrophil recruitment in the peritoneum, reduced histamine production and impaired bacterial clearence (PubMed:9862668). In experimental model of immune complex-mediated glomerulonephritis, reduced neutrophil accumulation without affecting their initial recruitment and reduced proteinuria (PubMed:9382884). Enhanced susceptibility to high fat diet-induced obesity characterized by a weight increase and higher levels of white and brown fat (PubMed:9207125). Reduced apoptosis in the hippocampus in neonates (PubMed:18685038). : P05555\n",
      "\n",
      "\n",
      "disease\n",
      "Disease susceptibility may be associated with variations affecting the gene represented in this entry. : P11215\n",
      "\n",
      "\n",
      "induction\n",
      "Induced by lipopolysaccharide (LPS) in mast cells. : P05555\n",
      "\n",
      "\n",
      "developmentalstage\n",
      "Strongly up-regulated in differentiating fetal muscle cells (in vitro). : Q9UKX5\n",
      "\n",
      "\n",
      "Keywords\n",
      "3D-structure : P20701, P11215, P05555, P20702\n",
      "Alternative splicing : P20701, P11215, P05555, Q9UKX5, O75578\n",
      "Calcium : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Cell adhesion : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Cell membrane : P20701, P61625, P24063, P11215, P05555\n",
      "Direct protein sequencing : P20701, P24063, P11215, P05555, P20702\n",
      "Disulfide bond : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Glycoprotein : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Host-virus interaction : P20701\n",
      "Integrin : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Magnesium : P20701, P61625, P24063, P11215, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Membrane : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Metal-binding : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Phagocytosis : P20701, P61625, P24063\n",
      "Phosphoprotein : P20701, P61625, P24063\n",
      "Polymorphism : P20701, P11215, P20702, Q9UKX5, O75578\n",
      "Receptor : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Reference proteome : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Repeat : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Signal : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Transmembrane : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Transmembrane helix : P20701, P61625, P24063, P11215, P05555, Q9QXH4, P20702, Q9QYE7, Q3V0T4, Q13349, Q9UKX5, O75578, P61622\n",
      "Immunity : P11215\n",
      "Innate immunity : P11215\n",
      "Systemic lupus erythematosus : P11215\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "\n",
    "blast_matches = []\n",
    "\n",
    "for accession in blast_uniprot_id:\n",
    "    path = os.path.join(wd, \"data/homologue\", gene,\"uniprot\",\"{0}.xml\".format(accession))\n",
    "    blast_matches.append(SeqIO.read(path, format=\"uniprot-xml\"))\n",
    "\n",
    "comments = {}\n",
    "an_domains = {}\n",
    "keywords = {}\n",
    "for match in blast_matches:\n",
    "    if \"keywords\" in match.annotations.keys():\n",
    "        for keyword in match.annotations[\"keywords\"]:\n",
    "            if keyword in keywords:\n",
    "                keywords[keyword] = \", \".join([keywords[keyword],match.id])\n",
    "            else:\n",
    "                keywords[keyword] = match.id\n",
    "    for annotation in match.annotations.keys():\n",
    "        if re.match(\"(comment_)[a-z]*\", annotation):\n",
    "            comment_type = annotation.split(\"_\",1)[1]\n",
    "            if comment_type not in comments.keys():\n",
    "                comments[comment_type] = {}\n",
    "            comment = match.annotations[annotation][0]\n",
    "            if comment in comments[comment_type].keys():\n",
    "                comments[comment_type][comment] = \", \".join([comments[comment_type][comment], match.id])\n",
    "            else:\n",
    "                 comments[comment_type][comment] = match.id\n",
    "\n",
    "for comment_type in comments.keys():\n",
    "    print(comment_type)\n",
    "    for comment in comments[comment_type].keys():\n",
    "        print(comment, \":\", comments[comment_type][comment])\n",
    "    print(\"\\n\")\n",
    "\n",
    "print(\"Keywords\")\n",
    "for keyword in keywords.keys():\n",
    "    print(keyword, \":\", keywords[keyword])\n",
    "print(\"\\n\")"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "As anotações confirmam a relação das proteínas homólogas e o suporte ao sistema imunitário, indicando que esta função é conservada entre as espécies de mamíferos, portadores deste gene.\n",
    "\n",
    "### Features"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "topological domain\n",
      "[25:1090] Extracellular : P20701\n",
      "[1111:1170] Cytoplasmic : P20701\n",
      "[23:1084] Extracellular : P61625\n",
      "[1105:1165] Cytoplasmic : P61625\n",
      "[23:1087] Extracellular : P24063\n",
      "[1108:1163] Cytoplasmic : P24063\n",
      "[16:1104] Extracellular : P11215\n",
      "[1128:1152] Cytoplasmic : P11215\n",
      "[16:1105] Extracellular : P05555\n",
      "[1129:1153] Cytoplasmic : P05555\n",
      "[19:1116] Extracellular : Q9QXH4\n",
      "[1137:1169] Cytoplasmic : Q9QXH4\n",
      "[19:1107] Extracellular : P20702\n",
      "[1128:1163] Cytoplasmic : P20702\n",
      "[19:1100] Extracellular : Q9QYE7\n",
      "[1121:1161] Cytoplasmic : Q9QYE7\n",
      "[26:1107] Extracellular : Q3V0T4\n",
      "[1128:1168] Cytoplasmic : Q3V0T4\n",
      "[17:1099] Extracellular : Q13349\n",
      "[1120:1161] Cytoplasmic : Q13349\n",
      "[22:1141] Extracellular : Q9UKX5, P61622\n",
      "[1164:1188] Cytoplasmic : Q9UKX5, P61622\n",
      "[22:1122] Extracellular : O75578\n",
      "[1145:1167] Cytoplasmic : O75578\n",
      "\n",
      "\n",
      "transmembrane region\n",
      "[1090:1111] Helical : P20701\n",
      "[1084:1105] Helical : P61625\n",
      "[1087:1108] Helical : P24063\n",
      "[1104:1128] Helical : P11215\n",
      "[1105:1129] Helical : P05555\n",
      "[1116:1137] Helical : Q9QXH4\n",
      "[1107:1128] Helical : P20702, Q3V0T4\n",
      "[1100:1121] Helical : Q9QYE7\n",
      "[1099:1120] Helical : Q13349\n",
      "[1141:1164] Helical : Q9UKX5, P61622\n",
      "[1122:1145] Helical : O75578\n",
      "\n",
      "\n",
      "domain\n",
      "[155:327] VWFA : P20701\n",
      "[152:324] VWFA : P61625\n",
      "[152:325] VWFA : P24063\n",
      "[149:328] VWFA : P11215\n",
      "[163:338] VWFA : P05555\n",
      "[151:330] VWFA : Q9QXH4\n",
      "[164:339] VWFA : P20702\n",
      "[151:334] VWFA : Q9QYE7\n",
      "[158:341] VWFA : Q3V0T4\n",
      "[149:332] VWFA : Q13349\n",
      "[163:345] VWFA : Q9UKX5, P61622\n",
      "[166:350] VWFA : O75578\n",
      "\n",
      "\n",
      "calcium-binding region\n",
      "[467:476] : P20701\n",
      "[529:538] : P20701, P20702\n",
      "[589:598] : P20701\n",
      "[464:473] : P61625, P11215, P05555, Q13349\n",
      "[526:535] : P61625\n",
      "[586:595] : P61625\n",
      "[465:474] : P24063, P20702\n",
      "[527:536] : P24063\n",
      "[587:596] : P24063\n",
      "[528:537] : P11215, P05555, Q13349\n",
      "[591:600] : P11215, P05555, Q13349\n",
      "[466:475] : Q9QXH4, Q9QYE7\n",
      "[530:539] : Q9QXH4, Q9QYE7\n",
      "[593:602] : Q9QXH4, Q9QYE7\n",
      "[592:601] : P20702\n",
      "[473:482] : Q3V0T4\n",
      "[537:546] : Q3V0T4\n",
      "[600:609] : Q3V0T4\n",
      "[487:496] : Q9UKX5, P61622\n",
      "[550:559] : Q9UKX5, P61622\n",
      "[612:621] : Q9UKX5, P61622\n",
      "[493:502] : O75578\n",
      "[557:566] : O75578\n",
      "[619:628] : O75578\n",
      "\n",
      "\n",
      "short sequence motif\n",
      "[1114:1119] GFFKR motif : P20701\n",
      "[1110:1115] GFFKR motif : P61625, P24063\n",
      "[1130:1135] GFFKR motif : P11215, P20702\n",
      "[1131:1136] GFFKR motif : P05555\n",
      "[1139:1144] GFFKR motif : Q9QXH4\n",
      "[1125:1130] GFFKR motif : Q9QYE7, Q13349\n",
      "[1132:1137] GFFKR motif : Q3V0T4\n",
      "\n",
      "\n",
      "disulfide bond\n",
      "[72:80] : P20701\n",
      "[110:129] : P20701\n",
      "[652:707] : P20701\n",
      "[770:777] : P20701, P20702\n",
      "[844:861] : P20701\n",
      "[997:1013] : P20701\n",
      "[1020:1052] : P20701\n",
      "[70:78] : P61625\n",
      "[107:126] : P61625, P24063, Q9QXH4, P20702, Q9QYE7\n",
      "[649:704] : P61625\n",
      "[767:774] : P61625, Q13349\n",
      "[841:858] : P61625\n",
      "[993:1009] : P61625\n",
      "[1016:1048] : P61625, P24063\n",
      "[69:77] : P24063\n",
      "[146:199] : P24063\n",
      "[650:705] : P24063\n",
      "[766:773] : P24063\n",
      "[839:856] : P24063\n",
      "[992:1009] : P24063\n",
      "[65:73] : P11215, P05555\n",
      "[104:123] : P11215, P05555\n",
      "[653:711] : P11215, P05555\n",
      "[769:776] : P11215, P05555, Q9QXH4\n",
      "[846:864] : P11215\n",
      "[997:1022] : P11215, P20702\n",
      "[1026:1032] : P11215, P20702\n",
      "[998:1023] : P05555\n",
      "[1027:1033] : P05555\n",
      "[68:76] : Q9QXH4, P20702, Q9QYE7\n",
      "[115:146] : Q9QXH4\n",
      "[495:507] : Q9QXH4\n",
      "[639:721] : Q9QXH4\n",
      "[655:711] : Q9QXH4, Q9QYE7\n",
      "[857:873] : Q9QXH4\n",
      "[1006:1031] : Q9QXH4\n",
      "[1035:1041] : Q9QXH4\n",
      "[115:145] : P20702\n",
      "[494:506] : P20702\n",
      "[638:722] : P20702\n",
      "[654:712] : P20702\n",
      "[847:863] : P20702\n",
      "[768:775] : Q9QYE7\n",
      "[844:860] : Q9QYE7, Q13349\n",
      "[992:1017] : Q9QYE7, Q13349\n",
      "[1021:1027] : Q9QYE7, Q13349\n",
      "[75:83] : Q3V0T4, Q9UKX5, P61622\n",
      "[114:133] : Q3V0T4\n",
      "[662:718] : Q3V0T4\n",
      "[775:782] : Q3V0T4\n",
      "[851:867] : Q3V0T4\n",
      "[999:1024] : Q3V0T4\n",
      "[1028:1034] : Q3V0T4\n",
      "[66:74] : Q13349\n",
      "[105:124] : Q13349\n",
      "[653:709] : Q13349\n",
      "[120:139] : Q9UKX5, P61622\n",
      "[128:159] : Q9UKX5, P61622\n",
      "[658:668] : Q9UKX5, P61622\n",
      "[673:729] : Q9UKX5, P61622\n",
      "[780:787] : Q9UKX5, P61622\n",
      "[880:893] : Q9UKX5, P61622\n",
      "[75:86] : O75578\n",
      "[665:675] : O75578\n",
      "[680:736] : O75578\n",
      "[788:795] : O75578\n",
      "\n",
      "\n",
      "compositionally biased region\n",
      "[1152:1161] Poly-Leu : Q9UKX5\n",
      "[1172:1176] Poly-Arg : Q9UKX5\n",
      "[1133:1140] Poly-Leu : O75578\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "features = {}\n",
    "for match in blast_matches:\n",
    "    for feature in match.features:\n",
    "        if re.search(\"(domain|motif|bond|region)\", feature.type):\n",
    "            feature_type = feature.type\n",
    "            if feature_type not in features.keys():\n",
    "                features[feature_type] = {}\n",
    "\n",
    "            feature_desc = str(feature.location)\n",
    "            if \"description\" in feature.qualifiers.keys():\n",
    "                feature_desc = \" \".join([feature_desc,feature.qualifiers[\"description\"]])\n",
    "\n",
    "            if feature_desc in features[feature_type].keys():\n",
    "                features[feature_type][feature_desc] = \", \".join([features[feature_type][feature_desc], match.id])\n",
    "            else:\n",
    "                 features[feature_type][feature_desc] = match.id\n",
    "\n",
    "for feature_type in features.keys():\n",
    "    print(feature_type)\n",
    "    for feature in features[feature_type].keys():\n",
    "        print(feature, \":\", features[feature_type][feature])\n",
    "    print(\"\\n\")"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Apesar da varialibilidade da localização das *features*, expectável dado que estas não se encontram alinhadas, podemos perceber que estas são comuns entre os homólogos, identificando locais de interesse para estudo."
   ],
   "metadata": {
    "collapsed": false
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}